synapse/events/snapshot.py

```diff
@@ -141,7 +116,7 @@ def deserialize(store, input):
         Returns:
             EventContext
         """
-        context = EventContext(
+        context = AsyncEventContext(
```

Comment:
Doesn't it make more sense to have this `deserialize` method on `AsyncEventContext`, rather than have `EventContext.deserialize` always return a `AsyncEventContext`?

Reply:
yeah but then I'd have to change more lines of code! think of all that typing!

Reply:
thinking about this a little more:

I'm not sure it *does* make more sense to move this to AsyncEventContext. I don't think it should be the caller's responsibility to decide whether the whole of the EventContext is serialised or if some of it is instead pulled from the database: the caller should just be able to use a `serialize`/`deserialize` pair.

In other words, AsyncEventContext is an implementation detail which I don't think needs exposing outside this file.

Reply:
Hmm, OK. I wonder whether it becomes clearer if we rename it `_AsyncEventContextImpl` or something. I think at the moment it feels somewhat like we have an sync and async version to choose from, rather than an `EventContext` interface that may or may not be async. IDK.

Reply:
ok, done